src/main/java/thrift/logic/parser/ListCommandParser.java

```diff
@@ -18,25 +19,44 @@
      */
     public ListCommand parse(String args) throws ParseException {
         ArgumentMultimap argMultimap =
-                ArgumentTokenizer.tokenize(args, CliSyntax.PREFIX_MONTH, CliSyntax.PREFIX_TAG);
+                ArgumentTokenizer.tokenize(args, CliSyntax.PREFIX_MONTH);
+
+        if (!arePrefixesPresent(argMultimap, CliSyntax.PREFIX_MONTH)
+                && !args.isEmpty()) {
+            throw new ParseException(String.format(Messages.MESSAGE_INVALID_COMMAND_FORMAT,
+                    ListCommand.MESSAGE_USAGE)); //prefix m/ is not present so there are invalid arguments
+        }
+
         return getCommand(argMultimap);
     }
 
     private ListCommand getCommand(ArgumentMultimap argMultimap) throws ParseException {
-        Optional<String> month = argMultimap.getValue(CliSyntax.PREFIX_MONTH);
-        Optional<String> tag = argMultimap.getValue(CliSyntax.PREFIX_TAG);
-        if (month.isEmpty() && tag.isEmpty()) {
-            return new ListCommand(); //list all transactions
-        } else if (!month.isEmpty() && !tag.isEmpty()) {
-            return new ListCommand(); //filter by both month and tag, coming in v1.3
-        } else if (!month.isEmpty() && tag.isEmpty()) {
-            return new ListCommand(); //filter by month, coming in v1.3
-        } else if (month.isEmpty() && !tag.isEmpty()) {
-            return new ListCommand(); //filter by tag, coming in v1.3
-        } else { //bad args input or wrong prefixes used will throw the ParseException
-            throw new ParseException(
-                    String.format(Messages.MESSAGE_INVALID_COMMAND_FORMAT, ListCommand.MESSAGE_USAGE));
+        if (arePrefixesPresent(argMultimap, CliSyntax.PREFIX_MONTH)
+                && !argMultimap.getPreamble().isEmpty()) {
+            throw new ParseException(String.format(Messages.MESSAGE_INVALID_COMMAND_FORMAT,
+                    ListCommand.MESSAGE_USAGE)); //prefix m/ exists, but there is invalid preamble before the prefix.
+        } else if (arePrefixesPresent(argMultimap, CliSyntax.PREFIX_MONTH)) {
+            try {
+                Month month = ParserUtil.parseMonth(argMultimap.getValue(CliSyntax.PREFIX_MONTH).get()); //to be used
+            } catch (Exception e) {
+                throw new ParseException(String.format(Messages.MESSAGE_INVALID_MONTH_FORMAT,
+                        ListCommand.MESSAGE_MONTH));
+            }
```

Comment:
This `try-catch` block is redundant since `ParserUtil.parseMonth` will throw ParseException with the message and your method here throws ParseException` upwards.

Reply:
removed this trycatch block.